pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applicationlayer
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	ocsv1 "github.com/openshift/api/security/v1"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/render/common/statik"
+)
+
+const (
+	L7LogCollectorDeamonsetName = "l7-log-collector"
+	L7CollectorContainerName    = "l7-collector"
+	ProxyContainerName          = "envoy-proxy"
+	EnvoyLogsVolumeName         = "envoy-logs"
+	EnvoyConfigMapName          = "envoy-config"
+	EnvoyConfigMapKey           = "envoy-config.yaml"
+	APLName                     = "application-layer"
+)
+
+var log = logf.Log.WithName("applicationlayer")
+
+func ApplicationLayer(
+	pullSecrets []*corev1.Secret,
+	installation *operatorv1.InstallationSpec,
+	osType rmeta.OSType,
+	logsEnabled bool,
+	logIntervalSeconds *int64,
+	logRequestsPerInterval *int64,
+) render.Component {
+
+	return &applicationLayerComponent{
+		pullSecrets:            pullSecrets,
+		installation:           installation,
+		osType:                 osType,
+		logsEnabled:            logsEnabled,
+		logIntervalSeconds:     logIntervalSeconds,
+		logRequestsPerInterval: logRequestsPerInterval,
+	}
+}
+
+type applicationLayerComponent struct {
```

Comment:
Nit but the package name is applicationLayer so you could just name this `component` and it's clear that it's an applicationLayer component.

Reply:
fixed in 32a037380b877cb81e1d9e497672aeed296391a8